src/test/java/ui/tests/RegistrationTest.java

```diff
@@ -0,0 +1,82 @@
+package ui.tests;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import pageobjects.AccountCreationPage;
+import pageobjects.AccountPage;
+import pageobjects.AuthenticationPage;
+import pageobjects.MainPage;
+import ui.ConfigProperties;
+import ui.UserAccount;
+import ui.UserAccountRegistrationForm;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class RegistrationTest {
+    public static WebDriver driver;
+    public static MainPage mainPage;
+    public static AuthenticationPage authenticationPage;
+    public static AccountCreationPage accountCreationPage;
+    public static AccountPage accountPage;
+    private static final Logger logger = Logger.getLogger(RegistrationTest.class.getName());
+
+    @BeforeClass
+    public static void setup() {
+        logger.info("Setup process started");
+        System.setProperty("webdriver.chrome.driver", ConfigProperties.getProperty("chromedriver"));
+        logger.info("ChromeDriver initiated");
+        driver = new ChromeDriver();
+        logger.info("MainPage initiated");
+        mainPage = new MainPage(driver);
+        logger.info("AuthenticationPage initiated");
+        authenticationPage = new AuthenticationPage(driver);
+        logger.info("AccountCreationPage initiated");
+        accountCreationPage = new AccountCreationPage(driver);
+        logger.info("Account initiated");
+        accountPage = new AccountPage(driver);
+        logger.info("Setup implicitly wait for each method");
+        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+        driver.get(ConfigProperties.getProperty("mainpage"));
```

Comment:
It's probably better to keep general configuration in `@BeforeClass` but move pages manipulation to either `@Before` method or in test, so that it's clear what page is used in each test. Even though there's a single test for now :)

Reply:
Moved